src/openrct2/management/finance.c

```diff
@@ -76,6 +76,11 @@ void finance_payment(money32 amount, rct_expenditure_type type)
 	money32 new_money = cur_money - amount;
```

Comment:
You check for overflows below, but you may still cause it here.

Please move this assignment to the checks below to the new `else` clause which we know won't overflow.